pkg/services/chgm/chgm.go

```diff
@@ -50,32 +58,117 @@ type Service interface {
+	*investigation.Client
+}
+
+// Triggered will analyse chgm incidents that are newly created
+func (c *Client) Triggered() error {
+	res, err := c.investigateStoppedInstances()
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Sprintf("InvestigateInstances failed: %s\n", err.Error()))
+	}
+
+	fmt.Printf("the investigation returned %#v\n", res)
+
+	lsExists, err := c.LimitedSupportReasonsExist(c.Cluster.ID())
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Errorf("failed to determine if limited support reason already exists: %w", err).Error())
+	}
+
+	// if lsExists, silence alert and add investigation to notes
+	if lsExists {
+		fmt.Println("Unrelated limited support reason present on cluster, silencing")
+		return c.SilenceAlert(res.String() + "Unrelated limited support reason present on cluster, silenced.")
+	}
+
+	if res.UserAuthorized {
+		fmt.Println("The node shutdown was not the customer. Should alert SRE")
+		return c.UpdateAndEscalateAlert(res.String())
+	}
+
+	// The node shutdown was the customer
+	// Put into limited support, silence and update incident notes
+	fmt.Printf("Sending limited support reason: %s\n", chgmLimitedSupport.Summary)
+	return utils.Retry(3, time.Second*2, func() error {
+		return c.PostLimitedSupport(res.String())
+	})
+}
+
+// Resolved will take appropriate action against a cluster whose CHGM incident has resolved.
+func (c *Client) Resolved() error {
+	res, err := c.investigateStartedInstances()
+
+	// The investigation encountered an error & never completed - alert Primary and report the error, retrying as many times as necessary to escalate the issue
+	if err != nil {
+		err = c.AddNote(fmt.Sprintf("Resolved investigation did not complete: %v\n", err.Error()))
+		if err != nil {
+			fmt.Println("could not update incident notes")
+		}
+		return utils.Retry(3, time.Second*2, func() error {
+			return c.CreateNewAlert(c.getInvestigationFailureAlert(err), c.GetServiceID())
+		})
+	}
+
+	// Investigation completed, but the state in OCM indicated the cluster didn't need investigation
+	if res.ClusterNotEvaluated {
+		fmt.Printf("Cluster has state '%s' in OCM, and so investigation is not need\n", res.ClusterState)
+		err = utils.Retry(3, time.Second*2, func() error {
+			return c.AddNote(res.String())
+		})
+		if err != nil {
+			fmt.Printf("Failed to add notes to incident: %s\n", res.String())
+		}
+	}
+
+	// Investigation completed, but the cluster has not been restored properly - keep in limited support & alert Primary, retrying as many times as necessary to escalate the issue
+	// potentially noisy
+	if res.Error != "" {
+		fmt.Printf("investigation completed, but cluster has not been restored properly\n")
+		err = utils.Retry(3, time.Second*2, func() error {
+			return c.AddNote(res.String())
+		})
+		if err != nil {
+			fmt.Printf("failed to add notes to incident: %s\n", res.String())
+		}
+
+		// TODO maybe add unrelated limited support check here instead of inside investigation
```

Comment:
I think it makes more sense to check on the higher level (investigation). 

**chore:** either remove the TODO or implement it ;)

Reply:
removed the comment for now